src/passes/Monomorphize.cpp

```diff
@@ -181,23 +397,159 @@ struct Monomorphize : public Pass {
       //       keep optimizing from the current contents as we go. It's not
       //       obvious which approach is best here.
       doMinimalOpts(func);
-      doMinimalOpts(refinedFunc);
+      doMinimalOpts(monoFunc.get());
 
       auto costBefore = CostAnalyzer(func->body).cost;
-      auto costAfter = CostAnalyzer(refinedFunc->body).cost;
+      auto costAfter = CostAnalyzer(monoFunc->body).cost;
+      // TODO: check for either a size decrease (always good) or a significant
+      //       speed increase (as a tiny one, in a huge function, can lead to
+      //       wasteful duplicated code)
```

Comment:
Why is code size decreasing always good? Do we add the monomorphized function only when we can entirely replace the original function?

Reply:
Good catch, yeah, this TODO makes more sense for the case we can entirely replace, which is *not* the case here. I revised it.